gcp-bundle/LICENSE

```diff
@@ -549,12 +549,6 @@ License: Apache 2 - https://www.apache.org/licenses/LICENSE-2.0
 
 --------------------------------------------------------------------------------
 
-Group: org.slf4j  Name: slf4j-api  Version: 1.7.36
```

Comment:
```
➜ cp /Users/fokko.driesprong/work/iceberg/gcp-bundle/build/libs/iceberg-gcp-bundle-1.8.0-SNAPSHOT.jar .
➜ unzip iceberg-gcp-bundle-1.8.0-SNAPSHOT.jar
➜ find . | grep -i slf                                                                                 
./io/grpc/netty/shaded/io/netty/util/internal/logging/LocationAwareSlf4JLogger.class
./io/grpc/netty/shaded/io/netty/util/internal/logging/Slf4JLoggerFactory$NopInstanceHolder.class
./io/grpc/netty/shaded/io/netty/util/internal/logging/Slf4JLoggerFactory.class
./io/grpc/netty/shaded/io/netty/util/internal/logging/Slf4JLogger.class
```

Reply:
I don't think we can consider https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/internal/logging/Slf4JLogger.java as slf4j-api provided. It's different package/class. 

The Netty `NOTICE` (https://github.com/netty/netty/blob/4.1/NOTICE.txt) mentions slf4j:

```
This product contains a modified portion of 'SLF4J', a simple logging
facade for Java, which can be obtained at:

  * LICENSE:
    * license/LICENSE.slf4j.txt (MIT License)
  * HOMEPAGE:
    * https://www.slf4j.org/

```
That should be in our `NOTICE` (I will update the `NOTICE` in this PR).

So, strictly speaking, the Slf4jLogger is not provided by slf4j-api.

I still think that we should remove slf4-api from `LICENSE` because `org.slf4j*` is not bundle.

Reply:
Yes, I fully agree. I should not review before my morning coffee. `slf4j` is not bundled, so it make sense to remove it from the license 👍